packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -0,0 +1,594 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+
+import 'basic.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+
+/// Object for specifying a focus traversal policy used for configuring a
+/// [DefaultFocusTraversal] widget.
+///
+/// See also:
+///
+///   * [DefaultFocusTraversal], a widget that imposes a traversal policy on the
+///     [Focus] nodes below it in the widget hierarchy.
+///   * [FocusNode], which is affected by the traversal policy.
+///   * [WidgetOrderFocusTraversalPolicy], a policy that relies on the widget
+///     creation order to describe the order of traversal.
+///   * [ReadingOrderTraversalPolicy], a policy that describes the order as the
+///     natural "reading order" for the current [Directionality].
+///   * [DirectionalFocusTraversalPolicyMixin] a mixin class that implements
+///     focus traversal in an axis direction.
+abstract class FocusTraversalPolicy {
+  /// Creates a FocusTraversalPolicy.
```

Comment:
dubious docs. consider using the abstract const constructor boilerplate instead.

Reply:
Done.